src/random.rs

```diff
@@ -1,8 +1,16 @@
-use rand::{self, weak_rng, XorShiftRng, Rand, Rng};
+use rand::{self, weak_rng, XorShiftRng, Rand, Rng, SeedableRng};
 use std::cell::RefCell;
 
 thread_local! {
-    static WEAK_RNG: RefCell<XorShiftRng> = RefCell::new(weak_rng());
+    static WEAK_RNG: RefCell<XorShiftRng> = RefCell::new({
+        let mut rng = weak_rng();
```

Comment:
I reckon we could use `thread_rng` (which is slightly higher quality?) to seed the weaker XorShiftRng. Probably doesn't make much difference though.